src/components/layout/hooks/useCurrentActiveMediaQuery.tsx

```diff
@@ -81,33 +81,20 @@ export const useCurrentActiveMediaQuery = (
     const [state, _setState] = React.useState<MediaType>(initialMediaQuery);
 
     React.useLayoutEffect(() => {
-        let mounted = true;
-
         const queries = new Queries(breakpointsMap);
 
         const setState = () => {
             _setState(queries.getCurrentActiveMedia());
         };
 
-        const onChange = () => {
-            if (!mounted) {
-                return;
-            }
-
-            setState();
-        };
-
-        queries.addListeners(onChange);
+        queries.addListeners(setState);
 
         setState();
 
         return () => {
-            mounted = false;
-            queries.removeListeners(onChange);
+            queries.removeListeners(setState);
         };
-        // don't support runtime breakpoint redefinition. Breakpoints defined only one at LayoutTheme
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []);
+    }, [breakpointsMap]);
```

Comment:
if user pass layout theme object like
```tsx
<ThemeProvider layout={{
  config: {
    breakpoints: {...}
  }
}} >...</ThemeProvider>
```
and `App` will have some state upper ThemeProvider it will cause this function to be constantly recalculated.

We need good reasons to be able to set breakpoints dynamically in runtime

Reply:
if `config` never changes, the user can move its creation outside the component. Otherwise, he can expect the changes to be taken into account.